Version Agnostic Updating

jwrb opened the pull request:
The scripts will now download the latest full release from GitHub, Regardless of Version.

This negates the need for manual updating of version numbers within all scripts.

Also switch to Curl for downloading the file since it looks nicer.

Please note; due to the fact that this uses the GitHub API. We have a limited amount of requests that we can make.
I will be implementing some logic to check if this is the case and notify the user.
Alternatively; we can poll GitHub for the rate to check we are able to submit a core request before polling for the version. This will avoid blocking on the server/client side

Commit:
Version Agnostic Updating
The scripts will now download the latest full release from GitHub, Regardless of Version.

This negates the need for manual updating of version numbers within all scripts.

Also switch to Curl for downloading the file since it looks nicer.

Please note; due to the fact that this uses the GitHub API. We have a limited amount of requests that we can make.
I will be implementing some logic to check if this is the case and notify the user.
Alternatively; we can poll GitHub for the rate to check we are able to submit a core request before polling for the version. This will avoid blocking on the server/client side

IngCr3at1on commented:
Untested ack

Commit:
Remove Debug URL Messages

jwrb commented:
This is a sample of what the script looks like when run
```bash
root@paycoin:~# ./UpdatePaycoinNode.sh 
### Changing to paycoind directory
### Stopping Paycoin Server
Paycoin server stopping
### Changing to home directory
### Downloading Paycoin Core v0.3.3.0
######################################################################## 100.0%
### Installing Paycoin Core v0.3.3.0
Archive:  linux64.zip
  inflating: paycoin-qt              
  inflating: paycoind                
### System will now reboot

Broadcast message from root@paycoin.test
	(/dev/pts/1) at 18:37 ...

The system is going down for reboot NOW!
```

Commit:
Add Version Checking
Include option to allow user to not reboot system (designed for advanced installs)

A flag to bypass both of these for automated systems would be useful too. Specifically; the version checking and the non-automatic rebooting.
Also fix a spelling error

MitchellCash commented on UpdatePaycoinNode.sh:
```diff
@@ -1,16 +1,48 @@
+
+if [[ "${xpy_version}" == "" ]]
+then
+    echo "### Paycoin Server Not Running, Unable to Check Version"
```
Can we please standardise the capitlisation to sentence case

MitchellCash commented on UpdatePaycoinNode.sh:
```diff
@@ -1,16 +1,48 @@
+if [[ "${xpy_version}" == "" ]]
+then
+    echo "### Paycoin Server Not Running, Unable to Check Version"
+    echo "### The latest is available version is v${version}"
```
Funny sentence structure :P

MitchellCash commented on UpdatePaycoinNode.sh:
```diff
@@ -1,16 +1,48 @@
+        exit 1
+    else
+        echo "### You are running Paycoin Core v#{xpy_version}"
+        echo "### The latest is available version is v${version}"
```
Again funny sentence structure :P

jwrb replied:
This has been rectified

MitchellCash replied:
Not yet it hasn't :P 

It's got one too many 'is', something like 'The latest available version is v${version}'  works better.

jwrb replied:
That is definitely an error on my part. Must have missed it (the brain is funny
like that)
I'll patch it and push it back up

On Fri, Nov 20, 2015 at 8:32 a.m., Mitchell Cash notifications@github.com
wrote:
In UpdatePaycoinNode.sh
[https://github.com/mitchellcash/PaycoinAutoNode/pull/13#discussion_r45445541] :

> +xpy_version=$(./paycoind getinfo 2> /dev/null | grep '"version"' | cut -d\" -f4 | sed 's/\v//g' >/dev/null)
> +# Check we are running the same version; otherwise; update
> +
> +if [[ "${xpy_version}" == "" ]]
> +then
> -    echo "### Paycoin Server Not Running, Unable to Check Version"
> -    echo "### The latest is available version is v${version}"
>   +else 
> -    if [[ "${version}" == "${xpy_version}" ]]
> -    then
> -        echo "### You are already running the latest version: v${version}"
> -        echo "### Exiting updater script."
> -        exit 1
> -    else
> -        echo "### You are running Paycoin Core v#{xpy_version}"
> -        echo "### The latest is available version is v${version}"

Not yet it hasn't :P

It's got one too many 'is', something like 'The latest available version is
v${version}' works better.

—
Reply to this email directly or view it on GitHub
[https://github.com/mitchellcash/PaycoinAutoNode/pull/13/files#r45445541] .[https://github.com/notifications/beacon/ABeWHZgFKcEcjeKAQCFpymP5mNixHe1pks5pHtIYgaJpZM4GkBrN.gif]

mitchellcash commented:
I'm having some issues.

- I can't seem to get it to check the version of the currently running Paycoin instance when using the update script. Keeps stating `### Paycoin Server Not Running, Unable to Check Version`

```
root@localhost:~# bash ./UpdatePaycoinNode.sh
### Changing to paycoind directory
### Paycoin Server Not Running, Unable to Check Version
### The latest is available version is vv0.3.3.0
### Downloading Paycoin Core v0.3.3.0
######################################################################## 100.0%
### Installing Paycoin Core v0.3.3.0
Archive:  linux64.zip
  inflating: paycoin-qt              
  inflating: paycoind                
### We recommend a system reboot to finish installation
### Would you like to reboot your system? n
### Please reboot your system for these changes to take effect
```

Commit:
Fix Version Checking Error
Fix an error where the version was piped to /dev/null before it was parsed.

This will fix the issue where the updater failed to check current running version and if the server was already running.

MitchellCash commented on UpdatePaycoinNode.sh:
```diff
@@ -1,16 +1,47 @@
+if [[ "${xpy_version}" == "" ]]
+then
+    echo "### Paycoin server not running, Unable to check version"
+    echo "### The latest is available version is v${version}"
```
Again funny sentence structure here. It's got one too many 'is', something like 'The latest available version is v${version}'  works better.

MitchellCash committed:
Update README.md with Description, Usage and an Example

Commit:
Fix Grammatical Error
Fix issue with sentence structure

jwrb commented:
@mitchellcash Fixed

Commit:
Add Command Line Arguments
The script now accepts multiple command line arguments.

-v will echo the currently available version
-r will force a reboot after the script has run
-f will force update Paycoin, whether it is the latest version of not

These flags make the script useful to run on a cron job for automatic updating each day.

jwrb commented:
I have no added command line arguments to the script. These are great when used in conjunction with Cron for automatic updating. The script no longer requires any user interaction when using certain flags.

Here's an excerpt of my shell when running the -f and -r flags. The binary is updated despite already being v0.3.3.0. The system is them force rebooted for good measure.

```bash
root@paycoin:~# bash ./UpdatePaycoinNode.sh -f -r
### Changing to paycoind directory
### The latest available version is vv0.3.3.0
### Stopping Paycoin server
Paycoin server stopping
### Downloading Paycoin Core v0.3.3.0
######################################################################## 100.0%
### Installing Paycoin Core v0.3.3.0
Archive:  linux64.zip
  inflating: paycoin-qt              
  inflating: paycoind                
### We recommend a system reboot to finish installation
### System will reboot in 5 seconds
```

Commit:
Add Libraries, Update README.md, Add Help Message
Update help message. Update readme

jwrb commented:
This now includes #14

mitchellcash commented:
Nice one, starting to look really good! Sorry I merged my README pull not thinking it now conflicts with this.

Now I do have a couple of nits but we can decide if I should merge now and address nits in a later pull.
Remove the duplicate v:
```
### The latest is available version is vv0.3.3.0
```

Also if paycoind is not running UpdatePaycoinNode.sh still seems to wants to update the software regardless:
```
### Paycoin server not running, Unable to check version
### The latest is available version is vv0.3.3.0
### Downloading Paycoin Core v0.3.3.0
```

Also squash and rebase :) Sorry for being a nuisance!

jwrb commented:
The duplicate V actually isn't in the code. Seems like one of the tools to trim
the extra V isn't actually running (version is returned with a v by default; but
we need it without)
That's also the issue with it not checking your version. I'll look into this

On Mon, Nov 30, 2015 at 8:27 a.m., mitchellcash/PaycoinAutoNode
<reply@reply.github.com>
wrote:
Nice one, starting to look really good! Sorry I merged my README pull not
thinking it now conflicts with this.

Now I do have a couple of nits but we can decide if I should merge now and
address nits in a later pull.
Remove the duplicate v:

### The latest is available version is vv0.3.3.0


Also if paycoind is not running UpdatePaycoinNode.sh still seems to wants to
update the software regardless:

### Paycoin server not running, Unable to check version
### The latest is available version is vv0.3.3.0
### Downloading Paycoin Core v0.3.3.0


Also squash and rebase :) Sorry for being a nuisance!

—
Reply to this email directly or view it on GitHub
[https://github.com/mitchellcash/PaycoinAutoNode/pull/13#issuecomment-160553499] .[https://github.com/notifications/beacon/ABeWHaR7NMUcD5fjFRPuZFscKYyaUgHZks5pLAACgaJpZM4GkBrN.gif]

jwrb commented:
Looks like you're missing sed. Try `sudo apt-get install sed`. I'll also add this to the Deploy script

Commit:
Include sed as a dependency
Install sed at the same time as other required libraries to ensure that Paycoin deploys and updates correctly

jwrb commented:
Please see #15

mitchellcash commented:
D'oh
